torch/csrc/api/include/torch/nn/modules/any.h

```diff
@@ -0,0 +1,215 @@
+#pragma once
+
+#include <torch/detail/any.h>
+
+#include <torch/csrc/autograd/variable.h>
+#include <torch/csrc/utils/memory.h>
+#include <torch/csrc/utils/variadic.h>
+
+#include <memory>
+#include <type_traits>
+#include <typeinfo>
+#include <utility>
+#include <vector>
+
+namespace torch {
+namespace detail {
+template <typename Head = void, typename... Tail>
+void check_not_lvalue_references() {
```

Comment:
note: This can also be implemented using `std::conjunction` and `std::disjunction` (see my comments below)